components/docked-composer/index.jsx

```diff
@@ -0,0 +1,168 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Docked Composer design pattern](https://lightningdesignsystem.com/components/docked-composer/) in React.
+// Based on SLDS v2.9.4
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import shortid from 'shortid';
+
+import { DOCKED_COMPOSER } from '../../utilities/constants';
+
+import Button from '../button';
+
+const propTypes = {
+	actionButton: PropTypes.node,
+	composerBody: PropTypes.node,
+	header: PropTypes.shape({
+		icon: PropTypes.node,
+		label: PropTypes.string,
+	}),
+	id: PropTypes.string,
+	isOpen: PropTypes.bool,
+};
```

Comment:
I'd recommend a `onRenderFooter` prop. You have no idea as to the alignment or contents, so I'd have the consumers pass in any formatting HTML like `slds-col_bump-left slds-text-align_right`